ext/mysqlnd/mysqlnd_auth.c

```diff
@@ -490,24 +487,15 @@ mysqlnd_auth_change_user(MYSQLND_CONN_DATA * const conn,
 		}
 	}
 	if (ret == PASS) {
-		char * tmp = NULL;
-		/* if we get conn->username as parameter and then we first free it, then estrndup it, we will crash */
```

Comment:
I don't know what this comment tried to warn me about, so I was a little hesitant of touching it, but I see no crashes or memory leaks.

Reply:
Can you add a `ZEND_ASSERT(conn->username.s != user && conn->passwd.s != passwd)` here to ensure it can't happen?

Reply:
You meant before the assignment, right?